File: src/main/java/org/spilth/Doc.java
package org.spilth;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties({"ec", "text"})
public class Doc {
    @JsonProperty("a")
    public String artifactId;

    @JsonProperty("g")
    public String groupId;

    public String id;

    public String latestVersion;

    @JsonProperty("p")
    public String packaging;

    public String repositoryId;
    public int versionCount;
    public String timestamp;


}

File: src/main/kotlin/org/spilth/models/Doc.kt
package org.spilth.models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties("ec", "text")
class Doc {
    @JsonProperty("a")
    var artifactId: String? = null

    @JsonProperty("g")
    var groupId: String? = null

    var id: String? = null

    var latestVersion: String? = null

    @JsonProperty("p")
    var packaging: String? = null

    var repositoryId: String? = null
    var versionCount: Int = 0
    var timestamp: String? = null
}
